src/SapNwRfc/Pooling/ISapPooledConnection.cs

```diff
@@ -8,6 +8,22 @@ namespace SapNwRfc.Pooling
     /// </summary>
     public interface ISapPooledConnection : IDisposable
     {
+        /// <summary>
+        /// Gets the metadata for the specified type name.
+        /// </summary>
+        /// <param name="typeName">The type name.</param>
+        /// <param name="cancellationToken">The cancellation token for cancelling the underlying operation.</param>
+        /// <returns>The metadata for the type name.</returns>
+        ISapTypeMetadata GetTypeMetadata(string typeName, CancellationToken cancellationToken = default);
```

Comment:
As mentioned earlier, I would not use a pooled connection to get this data. Thus I don't see a reason at this moment to add these 2 methods to `ISapPooledConnection`. As a pooled connections main purpose is to invoke functions in an API-like environment.